client/lib/interval/index.js

```diff
@@ -38,71 +29,36 @@ export default React.createClass( {
 			EVERY_THIRTY_SECONDS,
 			EVERY_MINUTE,
 		] ).isRequired,
-		pauseWhenHidden: PropTypes.bool,
-		children: PropTypes.element,
-	},
+	};
 
-	getDefaultProps: () => ( {
-		pauseWhenHidden: true,
-	} ),
+	tick = () => {
+		this.props.onTick();
+	};
 
-	getInitialState: () => ( {
-		id: null,
-	} ),
+	start = ( props = this.props ) => {
+		this.interval = setInterval( this.tick, props.period );
+	};
```

Comment:
Here I'm a fan of recursive `setTimeout()` to prevent accidental infinite loops.